src/ExceptionMechanism.php

```diff
@@ -26,18 +26,28 @@ final class ExceptionMechanism
      */
     private $handled;
 
+    /**
+     * @var array|null Arbitrary extra data that might help the user understand
+     *                 the error thrown by this mechanism
+     */
+    private $data;
+
     /**
      * Class constructor.
      *
-     * @param string $type    Unique identifier of this mechanism determining
-     *                        rendering and processing of the mechanism data
-     * @param bool   $handled Flag indicating whether the exception has been
-     *                        handled by the user (e.g. via try..catch)
+     * @param string     $type    Unique identifier of this mechanism determining
+     *                            rendering and processing of the mechanism data
+     * @param bool       $handled Flag indicating whether the exception has been
+     *                            handled by the user (e.g. via try..catch)
+     * @param array|null $data    Arbitrary extra data that might help the user
+     *                            understand the error thrown by this mechanism
+     * @psalm-param array<string, mixed>|null $data
      */
-    public function __construct(string $type, bool $handled)
+    public function __construct(string $type, bool $handled, array $data = null)
```

Comment:
Why having this argument to be `null` by default instead of an empty `array`? I think that if we want to leave it as-is, then we should make the argument `nullable` for consistency: it doesn't make sense that an argument cannot accept `null`, but its default value is `null`